pkg/kubelet/pod_workers.go

```diff
@@ -0,0 +1,104 @@
+/*
```

Comment:
Maybe not in this PR, but now that this has a nice API and is separate we should write unit tests for it :)